php/context/class-fieldmanager-context-submenu.php

```diff
@@ -198,6 +201,63 @@ public function url() {
 		}
 	}
 
+	/**
+	 * Register a new REST API endpoint to expose the field's data.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @param string $object_type Not used, but added since the overriden function has it declared.
+	 */
+	public function register_rest_field( $object_type = 'submenu' ) {
+		if (
+			did_action( 'rest_api_init' )
+			&& $this->fm->can_show_in_rest()
+		) {
+			register_rest_route(
+				FM_REST_API_NAMESPACE,
+				"/submenu-settings/{$this->fm->name}",
+				array(
+					'callback' => array( $this, 'rest_callback' ),
+				)
+			);
+		}
+	}
+
+	/**
+	 * Callback function for the REST repsonse.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return mixed The fm field data.
+	 */
+	public function rest_callback() {
+		// Get the data.
+		$data = $this->get_data( 0, $this->fm->name );
+
+		/**
+		 * Filters a single field's data to the REST API.
+		 *
+		 * @since 1.3.0
+		 *
+		 * @param array  $data The field data.
+		 * @param object $this The current context object.
+		 */
+		$data = apply_filters( 'fm_rest_get_submenu_' . $this->fm->name, $data, $this );
+
+		/**
+		 * Filters all data passed to the REST API.
+		 *
+		 * @since 1.3.0
+		 *
+		 * @param array  $data The field data.
+		 * @param array  $data The field name.
+		 * @param string $this The current context object.
+		 */
+		$data = apply_filters( 'fm_rest_get_submenu', $data, $this->fm->name, $this );
```

Comment:
Both the `fm_rest_get` and `fm_rest_get_submenu` filters are documented with "Filters all data passed to the REST API," but they appear to slightly more specific than "all" in practice.